src/Controller/SuppliersController.php

```diff
@@ -73,8 +74,10 @@ public function list(Request $request, int $page = 1)
 
         try {
             $results = $this->searchApi->list($page);
-        } catch (NotFoundException | PaginationException $e) {
-            throw new NotFoundHttpException('Page not found', $e);
+        } catch (Exception $e) {
+             // refresh page on 500 error
+             header('Location: ' . $_SERVER['REQUEST_URI']);
```

Comment:
Let's do this in symfony rather than raw PHP. https://symfony.com/doc/current/controller.html#redirecting

Reply:
No problem. I found this symfony code that'll replicate the same behavior as the PHP code:

`return $this->redirect($request->getUri());`

I hope that's okay? I'll make a seperate branch and PR for this.